test/test_main.py

```diff
@@ -91,23 +92,19 @@ class TestCharmEvents(CharmEvents):
 
         def cleanup():
             self._clear_unit_db()
-            self._clear_symlinks()
+            self._clear_hooks()
             CharmBase.on = CharmEvents()
         self.addCleanup(cleanup)
 
     @classmethod
-    def _clear_symlinks(cls):
-        r, _, files = next(os.walk(JUJU_CHARM_DIR / 'hooks'))
-        for f in files:
-            absolute_path = Path(r) / f
-            if absolute_path.name == 'install' and absolute_path.is_symlink():
-                if os.readlink(absolute_path) != cls.CHARM_PY_RELPATH:
-                    raise SymlinkTargetError(f'"{absolute_path.name}" link does not point to {cls.CHARM_PY_RELPATH}')
-            elif absolute_path.name.endswith('-storage-attached') and absolute_path.is_symlink():
-                if os.readlink(absolute_path) != 'install':
-                    raise SymlinkTargetError(f'"{absolute_path.name}" link does not point to "install"')
-            else:
-                absolute_path.unlink()
+    def _prepare_initial_hooks(cls):
+        shutil.copytree(Path(__file__).parent / 'test_hooks', JUJU_CHARM_DIR / 'hooks', symlinks=True)
```

Comment:
The new organization of test_hooks on that loose directory doesn't feel great. It's a loose directory with some symlinks pointing to a wrong place, which are copied over something else completely different which only makes sense for this one test.

How about dropping that test directory altogether, and simply creating the symlinks?

Reply:
https://github.com/canonical/operator/pull/63/files#diff-3e85841b8cff61e2a613aa9b0c63a081R100-R113